src/parameterizations/vertical/MOM_kappa_shear.F90

```diff
@@ -579,70 +722,164 @@ subroutine Calc_kappa_shear_vertex(u_in, v_in, h, T_in, S_in, tv, p_surf, kappa_
       do K=1,nz+1
-        kappa_2d(I,K,J2) = 0.0 ; tke_2d(I,K) = 0.0
+        kappa_vertex(I,J,K) = 0.0 ; tke_2d(I,K) = 0.0
       enddo
     endif ; enddo ! i-loop
 
     if (CS%VS_viscosity_bug) then
       do K=1,nz+1 ; do I=IsB,IeB
         tke_io(I,J,K) = G%mask2dBu(I,J) * tke_2d(I,K)
-        kv_io(I,J,K) = ( G%mask2dBu(I,J) * kappa_2d(I,K,J2) ) * CS%Prandtl_turb
+        kv_io(I,J,K) = ( G%mask2dBu(I,J) * kappa_vertex(I,J,K) ) * CS%Prandtl_turb
       enddo; enddo
     else
       do K=1,nz+1 ; do I=IsB,IeB
         tke_io(I,J,K) = tke_2d(I,K)
-        kv_io(I,J,K) = kappa_2d(I,K,J2) * CS%Prandtl_turb
+        kv_io(I,J,K) = kappa_vertex(I,J,K) * CS%Prandtl_turb
       enddo; enddo
     endif
-    if (J>=G%jsc) then ; do K=1,nz+1 ; do i=G%isc,G%iec
-      ! Set the diffusivities in tracer columns from the values at vertices.
-      kappa_io(i,j,K) = G%mask2dT(i,j) * 0.25 * &
-                        ((kappa_2d(I-1,K,J2m1) + kappa_2d(I,K,J2)) + &
-                         (kappa_2d(I-1,K,J2)   + kappa_2d(I,K,J2m1)))
-    enddo ; enddo ; endif
-
   enddo ! end of J-loop
 
+
+  do j=G%jsc,G%jec
+    if (CS%Kd_tracer_mean_answer_date < 20250304) then
+      ! This is a non-thickness weighted arithmetic mean, but it is not bitwise identical to the more general
+      ! version of the mean below the "else" that includes a mathematically equivalent form.
+      do K=1,nz+1 ; do i=G%isc,G%iec
+        ! Set the diffusivities in tracer columns from the values at vertices.
+        kappa_io(i,j,K) = G%mask2dT(i,j) * 0.25 * &
+                         ((kappa_vertex(I-1,J-1,K)+kappa_vertex(I,J,K)) +&
+                          (kappa_vertex(I-1,J,K)+kappa_vertex(I,J-1,K)))
+      enddo; enddo
+    else
+      do i=G%isc,G%iec
+        ! It doesn't matter what these values are so make them zero.
+        kappa_io(i,j,1) = 0.0
+        kappa_io(i,j,nz+1) = 0.0
+      enddo
+      if (CS%VS_GeometricMean) then
+        do K=2,nz ; do i=G%isc,G%iec
+          ! Set the diffusivities in tracer columns from the values at vertices.
+          !  The geometric mean is zero if any component is zero, hence the need
+          !  and sensitivity to a value of kappa_trunc in regions on boundaries of shear zones.
+          Kd_ll = max(kappa_vertex(I-1,J-1,K),CS%kappa_trunc) ! Diffusivity, lower left
+          Kd_ur = max(kappa_vertex(I,J,K),CS%kappa_trunc) ! Diffusivity, upper right
+          Kd_ul = max(kappa_vertex(I-1,J,K),CS%kappa_trunc) ! Diffusivity, upper left
+          Kd_lr = max(kappa_vertex(I,J-1,K),CS%kappa_trunc) ! Difusivity, lower right
+          if (CS%VS_ThicknessMean) then
+            weight_ll = hweight_ll(i,j,k)
+            weight_ur = hweight_ur(i,j,k)
+            weight_ul = hweight_ul(i,j,k)
+            weight_lr = hweight_lr(i,j,k)
+          endif
+
+          ! The following expression generalizes the geometric mean at tracer points, w/ or w/o thickness weighting:
+          kappa_io(i,j,K) = G%mask2dT(i,j) * &
```

Comment:
The rational power function `x**a` is much less efficient than `sqrt()`, which is often done in hardware.  Also while `sqrt()` can be used with appropriate dimensional rescaling, `x**a` can not, so as written this expression will break the dimensional consistency testing.

Please consider using separate expressions for `kappa_io` depending on whether `CS%VS_ThicknessMean` is true, with the false option being `kappa_io(i,j,K) = G%mask2dT(i,j) * sqrt(sqrt( (Kd_ll*Kd_ur) * (Kd_ul*Kd_lr) ))` and the .true. option including an unscaling of `Kd_ll`, etc., and a rescaling back into rescaled units outside of the outermost parentheses.